unit_tests/src/test_user_administration.cpp

```diff
@@ -223,4 +229,26 @@ TEST_CASE("user group administration")
             irods::experimental::defer_authentication, env.rodsHost, env.rodsPort, {test_user.name, env.rodsZone}};
         CHECK(clientLoginWithPassword(static_cast<RcComm*>(user_conn), test_user_prop.value.data()) == 0);
     }
+
+    SECTION("#7576: ensure no limitations on proxied groupadmin")
+    {
+        rodsEnv env{};
+        _getRodsEnv(env);
+
+        REQUIRE_NOTHROW(
+            adm::client::modify_user(conn, generic_user, adm::user_type_property{adm::user_type::groupadmin}));
+        REQUIRE(adm::client::type(conn, generic_user) == adm::user_type::groupadmin);
+
+        // Connect to iRODS without authenticating.
+        irods::experimental::client_connection user_conn{env.rodsHost,
+                                                         env.rodsPort,
+                                                         {env.rodsUserName, env.rodsZone}, // (proxy) rodsadmin user
+                                                         {generic_user.name, env.rodsZone}}; // groupadmin user
+
+        // Try to add a group as the proxied groupadmin user.
+        adm::group test_group{"bad_bad_water_group"};
+        REQUIRE_NOTHROW(adm::client::add_group(user_conn, test_group));
+        irods::at_scope_exit group_cleanup{
+            [&user_conn, &test_group] { CHECK_NOTHROW(adm::client::remove_group(user_conn, test_group)); }};
```

Comment:
groupadmins cannot remove groups

https://docs.irods.org/4.3.1/icommands/groupadmin/

Reply:
> I'm a little confused by the at_scope_exit appearing at the end of the scope here... Maybe this could just be the remove_group call, sans at_scope_exit?

Can do. Would it still be a CHECK_NOTHROW or does it get upgraded to a REQUIRE_NOTHROW?

Reply:
CHECK_NOTHROW is fine. Since it's the last thing in the test I suppose it doesn't matter much

Reply:
Done. Kept it as a CHECK.